tools/release-all/README.md

```diff
@@ -0,0 +1,95 @@
+                      'major' Bump the major part.
+                      'minor' Bump the minor part.
+                      'patch' Bump the patch part.
+  -L                List recognized repo names, then exit.
+                      This list shows you the proper order to go through
+                      when creating releases for the repos. Some repos
+                      have references to others, so stick to the listed order!
+  -P phase          Indicates which phase to run. Default: 'master'.
+                      'master'       Validate the master/main branches.
+                      'release'      Create the release branches, but don't push.
+                      'release-push' Create and push the release branches.
+                      'create-pr'    Create PR for release branches.
+                      'merge-pr'     Merge PR for release branches.
+                      'tag-release'  Tag the release.
+  -R repo_names     Comma-separated list of repo names to operate on.
+                    If unspecified, then all repos will be used.
+  -w workspace_dir  Name for working directory. Default: 'workingspace'
+```
+
+## Steps
+
+### Run the steps in this order
+
+
+
+> **Note:** You almost always want to use the -R option to focus the `phase` activity to a specific repo.
+
+0. **List Repos:** Get the ordered list of repo names to use with -R option in subsequent steps. This is referred to as `repo-list`
+    > **Pro tip** Keep this list in a separate window for easy viewing
+    ./release-all.sh -L
+
+1. **Check Vendoring:** For each repo's master/main branch; determine whether any of them need to be re-vendored.
+    > **Note:** Ensure each repo is error-free before proceeding to the next repo in `repo-list`
+
+    ```bash
+    For each repo in `repo-list`
+        ./release-all.sh -P master -R $repo
+    ```
+
+2. **Create Trial Release Branch:** Create the new release branch, merge master/main to that release branch, but don't push it yet. The point of this step is to look for merge conflicts between master/main and the release branch.
+
+    ```bash
+    For each repo in `repo-list`
+        ./release-all.sh -P release -R $repo
+    ```
+
+3. **Generate Release:** For each repo in `repo-list`, proceed through the following steps in sequence before moving on to the next repo.
+    > **Note:** The next steps use the gh(1) GitHub CLI tool and require a GH_TOKEN environment variable containing a 'repo' scope classic token.
+
+    1. If the **Create Trial Release Branch** had no errors
+      ./release-all.sh -P release-push -R <repo>
+
+        Else if **Create Trial Release Branch** was unable to auto merge, manually merge and push the release branch
+
+        ```bash
+        cd workingspace/repo
+        # Manually merge the changes from master/main to the release branch
+        go mod tidy
+        go mod vendor
+        git commit -s # take the default commit message, don't bother editing it.
```

Comment:
Before the commit run `git status` to confirm that all issues have been addressed.

Reply:
Also, this needs a `git add ....` of affected files.